src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -657,7 +675,7 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                             return task;
                         }
 
-                        goto case State.ResourceEnd;
+                        goto case State.InvokeEnd;
```

Comment:
I can't think of a way that this will manifest in observable bad behavior, but it's part of the code we're fixing and it's just so obviously wrong.

"if the task is completed goto state1; else await the task and then goto state2"

If you still feel strongly then I'm at least going to leave a comment here that this is wrong but we decided not to fix it, and then I'll fix it in dev.